src/org/thoughtcrime/securesms/ConversationListFragment.java

```diff
@@ -379,7 +380,32 @@ private void handleCreateConversation(int chatId) {
     ((ConversationSelectedListener)getActivity()).onCreateConversation(chatId);
   }
 
-  public void loadChatlist() {
+  private boolean inLoadChatlist;
+  private boolean needsAnotherLoad;
+  private void loadChatlistAsync() {
+    if (inLoadChatlist) {
+      needsAnotherLoad = true;
+      Log.i(TAG, "chatlist loading debounced");
+    } else {
+      inLoadChatlist = true;
+      Util.runOnAnyBackgroundThread(() -> {
+
+        needsAnotherLoad = false;
+        loadChatlist();
+
+        while (needsAnotherLoad) {
+          Util.sleep(100);
+          needsAnotherLoad = false;
+          Log.i(TAG, "executing debounced chatlist loading");
+          loadChatlist();
+        }
+
```

Comment:
in theory, here is a potential race condition: if the background thread is exactly here and the main thread checks `inLoadChatlist` at that moment, one update will be missing. that can probably be fixed with some other multi-theading techniques, any suggestions welcome :) however, in practice, these boolean may be pretty fine, for incremental search, we're doing the same, and i have not seen a race condition there in practise (would be easy noticable as the last typed character is not searched for)